addons/sale/models/analytic.py

```diff
@@ -128,11 +128,13 @@ def _sale_determine_order_line(self):
                 raise UserError(_('The Sales Order %s linked to the Analytic Account must be validated before registering expenses.') % sale_order.name)
 
             price = analytic_line._sale_get_invoice_price(sale_order)
-            so_line = self.env['sale.order.line'].search([
-                ('order_id', '=', sale_order.id),
-                ('price_unit', '=', price),
-                ('product_id', '=', self.product_id.id)
-            ], limit=1)
+            so_line = None
+            if analytic_line.product_id.expense_policy == 'sales_price' and analytic_line.product_id.invoice_policy == 'delivery':
```

Comment:
why was this changed exactly?
expense_policy is 'no' by default, and you can't change it unless you have the entire sale_expense and hr_expense modules installed, so by default this always seems to cause it to generate new lines, even when you just want to create a credit note for the return of a product on the sale order